[REEF-1438] Validate Exception in spun off System.Threading.Tasks.Task does not trigger FailedEvaluator event

tcNickolas opened the pull request:
JIRA:
  [REEF-1438](https://issues.apache.org/jira/browse/REEF-1438)

Pull request:
  This closes #

tcNickolas commented:
This change also cleans up several tests which I've looked at while adding a new one. 

A good AppVeyor build (for at least one job) can be found at https://ci.appveyor.com/project/tcNickolas/reef/build/484-REEF-1438

markusweimer commented:
I left a question over on JIRA.

jwang98052 commented on lang/cs/Org.Apache.REEF.Tests/Functional/Failure/User/UnhandledTaskExceptionInTaskTest.cs:
```diff
@@ -0,0 +1,231 @@
+
+                    value.GetActiveContext().Value.Dispose();
+
+                    if (_numFailedTasksReceived == 2)
```
Put the log before context dispose to make sure it is logged before driver is shut down.

jwang98052 commented:
I completed a pass.

tcNickolas commented:
I've addressed the comment and updated the PR. Apache AppVeyor seems to be stuck again, a good build is at my branch: https://ci.appveyor.com/project/tcNickolas/reef/build/486-REEF-1438

jwang98052 commented:
LGTM. I will merge it.
